cmd/launcher/watchdog/watchdog_service_windows.go

```diff
@@ -0,0 +1,230 @@
+	time.Sleep(time.Second)
+
+	return nil
+}
+
+func (w *winWatchdogSvc) Execute(args []string, r <-chan svc.ChangeRequest, changes chan<- svc.Status) (ssec bool, errno uint32) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	const cmdsAccepted = svc.AcceptStop | svc.AcceptShutdown
+	changes <- svc.Status{State: svc.StartPending}
+	w.slogger.Log(ctx, slog.LevelInfo, "executing windows service")
+	changes <- svc.Status{State: svc.Running, Accepts: cmdsAccepted}
+
+	runRestartServiceResults := make(chan struct{})
+
+	gowrapper.Go(ctx, w.systemSlogger.Logger, func() {
+		if err := runLauncherWatchdogService(ctx, w); err != nil {
+			w.systemSlogger.Log(ctx, slog.LevelInfo,
+				"runLauncherWatchdogService exited",
+				"err", err,
+				"stack_trace", fmt.Sprintf("%+v", errors.WithStack(err)),
+			)
+		} else {
+			w.systemSlogger.Log(ctx, slog.LevelInfo,
+				"runLauncher exited cleanly",
+			)
+		}
+
+		// signal to fully exit so that the service manager can restart the service
+		runRestartServiceResults <- struct{}{}
+	}, func(r any) {
+		w.systemSlogger.Log(ctx, slog.LevelError,
+			"exiting after runLauncherWatchdogService panic",
+			"err", r,
+		)
+
+		// signal to fully exit so that the service manager can restart the service.
+		runRestartServiceResults <- struct{}{}
+	})
+
+	for {
+		select {
+		case c := <-r:
+			switch c.Cmd {
+			case svc.Interrogate:
+				changes <- c.CurrentStatus
+				// Testing deadlock from https://code.google.com/p/winsvc/issues/detail?id=4
+				time.Sleep(100 * time.Millisecond)
+				changes <- c.CurrentStatus
+			case svc.Stop, svc.Shutdown:
+				w.systemSlogger.Log(ctx, slog.LevelInfo, "shutdown request received")
+				changes <- svc.Status{State: svc.StopPending}
+				cancel()
+				time.Sleep(1 * time.Second) // give checker routine enough time to shut down
+				changes <- svc.Status{State: svc.Stopped, Accepts: cmdsAccepted}
+				return ssec, errno
+			default:
+				w.systemSlogger.Log(ctx, slog.LevelInfo,
+					"unexpected change request",
+					"service", launcherWatchdogServiceName,
+					"change_request", fmt.Sprintf("%+v", c),
+				)
+			}
+		case <-runRestartServiceResults:
+			w.systemSlogger.Log(ctx, slog.LevelInfo,
+				"shutting down restart service after exit",
+			)
+			// We don't want to tell the service manager that we've stopped on purpose,
+			// so that the service manager will restart launcher correctly.
+			// We use this error code largely because the windows/svc code also uses it
+			// and it seems semantically correct enough; it doesn't appear to matter to us
+			// what the code is.
+			return false, uint32(windows.ERROR_EXCEPTION_IN_SERVICE)
+		}
+	}
+}
+
+func (w *winWatchdogSvc) checkLauncherStatus(ctx context.Context) error {
+	serviceManager, err := mgr.Connect()
+	if err != nil {
+		w.slogger.Log(ctx, slog.LevelError,
+			"connecting to service control manager",
+			"err", err,
+		)
+
+		return err
```

Comment:
```suggestion
		return fmt.Errorf("connecting to service control manager: %w", err)
```

since the calling function will log the error